java/compaction-core/src/main/java/sleeper/compaction/job/status/CompactionJobStatus.java

```diff
@@ -138,22 +138,48 @@ private Instant firstTime() {
 
     public Instant lastTime() {
         if (isFinished()) {
-            return finishedStatus.getUpdateTime();
+            return getLatestFinishedStatus().getUpdateTime();
         } else if (isStarted()) {
-            return startedStatus.getUpdateTime();
+            return getLatestStartedStatus().getUpdateTime();
         } else {
             return createdStatus.getUpdateTime();
         }
     }
```

Comment:
I think a lot of these functions will produce unexpected behaviour when there's a finished run followed by an unfinished run. You'd expect the start and finish times to refer to the same run, but they won't in that case.

It might be clearer if we move the relevant functions to the run rather than the job status.

Can we make sure this case is tested, and for reporting and for loading the updates from DynamoDB?

Reply:
Never mind, we can fix this separately in issue [261](https://github.com/gchq/sleeper/issues/261) since the problem is that these methods don't really make sense given that there are multiple runs (particularly isFinished when not all runs are finished).